service_discovery/README.md

```diff
@@ -0,0 +1,196 @@
+# Service Discovery
+
+- [Introduction](#introduction)
+- [High-level Design](#high-level-design)
+- [Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Building](#building)
+  - [Running the Tests](#running-the-tests)
+- [Running the Service](#running-the-service)
+  - [Sample Applications](#sample-applications)
+- [Developing with Service Discovery](#developing-with-service-discovery)
+- [Trademarks](#trademarks)
+
+## Introduction
+
+This Service Discovery system for Eclipse Chariott aims to simplify in-vehicle application development by abstracting out where services are running, so that applications that want to leverage the resources and capabilities of another application can discover the location and metadata necessary to communicate. This is a new version of Service Discovery which is decoupled from the intent broker, and it is currently under development. If you would like to use the existing version of Chariott with intent brokering, please refer to the [top-level Readme](./../README.md) for details.
+
+The [Getting Started](#getting-started) section shows how to get started and run the Service Discovery system locally.
+
+The [Developing with Service Discovery](#developing-with-service-discovery) shows what is necessary to start developing applications that use Service Discovery.
+
+## High-level Design
+
+The service discovery system consists of 3 parts:
+
+1. The Service Registry
+2. A provider
```

Comment:
We already have components that also use "provider" and "consumer", Let's try to have distinctive names for each.

Reply:
Going with "service" for now from a service discovery perspective.
From a document on service discovery:
Service discovery has three components.

A service provider; that provides the services across a network.
A service registry; that is a database containing the locations of all available service instances.
A service consumer; who retrieves the service provider’s location from the service registry and communicates with the service instance.

Reply:
Much better.